ibazel/live_reload/server.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package live_reload
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"strconv"
+
+	blaze_query "github.com/bazelbuild/bazel-watcher/third_party/bazel/master/src/main/protobuf"
+	"github.com/jaschaephraim/lrserver"
+)
+
+var noLiveReload = flag.Bool("nolive_reload", false, "Disable JavaScript live reload support")
+
+type LiveReloadServer struct {
+	lrserver *lrserver.Server
+}
+
+func New() *LiveReloadServer {
+	return &LiveReloadServer{}
+}
+
+func (l *LiveReloadServer) Setup() {}
+
+func (l *LiveReloadServer) startLiveReloadServer() {
+	if l.lrserver != nil {
+		return
+	}
+
+	port := lrserver.DefaultPort
+	for ; port < lrserver.DefaultPort+100; port++ {
+		if testPort(port) {
+			l.lrserver = lrserver.New("live reload", port)
+			// Live reload server shouldn't log.
+			l.lrserver.SetStatusLog(log.New(os.Stderr, "", 0))
+			go func() {
+				err := l.lrserver.ListenAndServe()
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Live reload server failed to start: %v\n", err)
+				}
+			}()
+			url := fmt.Sprintf("http://localhost:%d/livereload.js?snipver=1", port)
+			os.Setenv("IBAZEL_LIVERELOAD_URL", url)
+			return
+		}
+	}
+	fmt.Fprintf(os.Stderr, "Could not find open port for live reload server\n")
+}
+func (l *LiveReloadServer) Cleanup() {}
```

Comment:
Can you leave a comment as to why its not necessary to kill the reload server and unset the IBAZEL_LIVERELOAD_URL within the cleanup step? I didn't expect this method to be empty

Reply:
Good eyes. I messed that one up real good!